testing/upload_validation.ts

```diff
@@ -227,6 +228,120 @@ export function validateSyncTableSchema(
   );
 }
 
+/**
+ * Returns a map of sync table names to their child sync table names, or undefined if the hierarchy is invalid.
+ * Example valid return: { Parent: 'Child' }
+ * {} is also a valid result, when there are no sync tables, or no parent relationships.
+ * @hidden
+ */
+export function validateCrawlHierarchy(
+  pack: PackVersionDefinition,
+  context?: z.RefinementCtx,
+): Record<string, string[]> | undefined {
+  const result: Record<string, string[]> = {};
```

Comment:
Maybe a name like `parentToChildrenTableNames` could be more clear than `result`? It's a large enough function that the docstring might not always be visible as people are reading through it.

Reply:
Great point.